### PR TITLE
feat(linter): add jsx-a11y/no-noninteractive-tabindex rule

### PR DESCRIPTION
**Pull request by taearls**
This PR implements the [jsx-a11y/no-noninteractive-tabindex](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/91e39b45ade789c86ae14df869a86b0ea468ed95/docs/rules/no-noninteractive-tabindex.md) rule. 

I set the default values as the ones that are in the recommended config.

I used this [html spec section](https://html.spec.whatwg.org/multipage/dom.html#interactive-content) to source the interactive elements.

The interactive roles came from [this article in the MDN docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles).

### CODE REVIEW

File: crates/oxc_linter/src/rules/jsx_a11y/no_noninteractive_tabindex.rs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+impl Rule for NoNoninteractiveTabindex {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::JSXOpeningElement(jsx_el) = node.kind() else {
+            return;
+        };
+
+        if let Some(JSXAttributeItem::Attribute(tabindex_attr)) =
+            has_jsx_prop_ignore_case(jsx_el, "tabIndex")
+        {
+            if let Some(JSXAttributeValue::StringLiteral(tabindex)) = &tabindex_attr.value {
```
Comment (camc314) [inline crates/oxc_linter/src/rules/jsx_a11y/no_noninteractive_tabindex.rs:100]: lets flatten these into let - else stmts
```suggestion
        let Some(JSXAttributeItem::Attribute(tabindex_attr)) = has_jsx_prop_ignore_case(jsx_el, "tabIndex") else { return; };
        let Some(JSXAttributeValue::StringLiteral(tabindex)) = &tabindex_attr.value else { return; };
```

as it's currently much more nested than it needs to be

Comment (taearls) [reply crates/oxc_linter/src/rules/jsx_a11y/no_noninteractive_tabindex.rs]: done

**Comment by taearls** [comment]
I added comments regarding the interactive html roles and elements, because it's convenient to have linked resources to relevant official spec sections, and specifically an explanation regarding the tabpanel role. If this is not desired I can remove.

**Review by camc314 — APPROVED** [review]
thanks!

**Pull request merged**